ddtrace/tracer/span.go

```diff
@@ -77,6 +77,8 @@ type span struct {
 	noDebugStack bool         `msg:"-"` // disables debug stack traces
 	finished     bool         `msg:"-"` // true if the span has been submitted to a tracer.
 	context      *spanContext `msg:"-"` // span propagation context
+	inProcessor  bool         `msg:"-"` // true if the span is in postProcessor.
```

Comment:
I don't understand the point of this field. I'm pretty sure we can get rid of it and use only `finished`

Reply:
I made the distinction between `inProcessor` and `finished` because `(s *span) finish(finishTime int64) bool` should never be called on a finished span, even if it is in the processor (vs being able to call `SetTag` or `SetOperationName` on a finished span if it is in the processor). If I don't make the distinction, I believe I would need to unfinish the spans (`span.finished = false`), which will enable calling `(s *span) finish(finishTime int64) bool` on a finished span. Does that make sense ? Maybe there is a better way to do this ?

Reply:
Can't we run the processor before `s.finished = true` ? Would there be any harm in that?